### ISSUE TITLE
logging changes between 1.4 and 2.0

### ISSUE THREAD

**Issue by fooelisa** [Juniper/ansible-junos-stdlib#354]
Hi, after an upgrade from 1.4 and 2.0 I've noticed some changes in the logging behavior. I am not sure about the feature to configure the log level though Ansible's verbosity. I'd like for the router log file to be more chatty than ansible and what the user has to see on the CLI. I can't depend on the user to always remember to attach the required number of v's and there doesn't seem to be a way of permanently setting the log level for ansible at the moment either. Did you guys have anything in mind for such a scenario? 

Thanks
Elisa

**Comment by stacywsmith** [comment]
In my opinion (remembering I no longer speak for Juniper) this is a reasonable and worthwhile request.

I believe this could be implemented fairly easily with:
1) Specify an additional logging-specific option, such as `level`, [here](https://github.com/Juniper/ansible-junos-stdlib/blob/master/module_utils/juniper_junos_common.py#L497-L501) and [here](https://github.com/Juniper/ansible-junos-stdlib/blob/master/module_utils/juniper_junos_common.py#L323-L384)
I would suggest that the `level` argument could use Ansible's `choices` functionality to restrict the value to only known logging levels.
 
2) Use the value of the new option to set the logging level [here](https://github.com/Juniper/ansible-junos-stdlib/blob/master/module_utils/juniper_junos_common.py#L868-L876).

**Comment by vnitinv** [comment]
These were some changes done in current github code https://github.com/Juniper/ansible-junos-stdlib/pull/337/files which is not present in galaxy. 
@fooelisa from were did you install roles? So if you install from github and provide logfile option, all the roles related logs should go in that given logfile.

**Comment by fooelisa** [comment]
Got the latest github running, not using galaxy. They go into the logfile alright, configuring the verbosity is the issue.

**Comment by rsmekala** [comment]
@fooelisa Could you try the following command:

`export ANSIBLE_DEBUG=True`

This will set the logging level of both Ansible and Juniper.junos module to DEBUG level. The fix has been mentioned in [Ansible Docs](https://docs.ansible.com/ansible/2.4/network_debug_troubleshooting.html#enabling-networking-logging-and-how-to-read-the-logfile)

Let us know the result.

**Comment by fooelisa** [comment]
That surely works, but this sets *all* of Ansible to DEBUG mode. That means the user gets 1000's of lines of convoluted output on the CLI, which is what I am trying to avoid. Unfortunately Ansible doesn't provide a way to set the level to INFO, which would a compromise I could live with.

**Comment by rsmekala** [comment]
@fooelisa In sync with @stacywsmith , we would like to implement something like:

```
tasks:
    - name: Gather Junos facts with no configuration
      juniper_junos_facts:
        level: DEBUG
```

The user can choose out of INFO/DEBUG logging level using this option. The default logging level will be set to WARNING.

The order of preference for setting the level would be:
1. Set level to DEBUG if ANSIBLE_DEBUG is True.
2. Set level based on number of v's passed to module.
3. Set level based on level option mentioned in task data.

Could you please confirm this is what you are looking for ??

**Comment by fooelisa** [comment]
Yep, that sounds great! Ty

**Comment by vnitinv** [comment]
Fixed via #359
